test/e2e/k8s_util.go

```diff
@@ -81,25 +82,40 @@ func (k *KubernetesUtils) GetPods(ns string, key string, val string) ([]v1.Pod,
 
 // Probe execs into a Pod and checks its connectivity to another Pod.  Of course it assumes
 // that the target Pod is serving on the input port, and also that ncat is installed.
-func (k *KubernetesUtils) Probe(ns1, pod1, ns2, pod2 string, port int32) (bool, error) {
-	fromPods, err := k.GetPods(ns1, "pod", pod1)
-	if err != nil {
-		return false, errors.WithMessagef(err, "unable to get pods from ns %s", ns1)
-	}
-	if len(fromPods) == 0 {
-		return false, errors.New(fmt.Sprintf("no pod of name %s in namespace %s found", pod1, ns1))
-	}
-	fromPod := fromPods[0]
+func (k *KubernetesUtils) Probe(ns1, pod1, ns2, pod2 string, port int32, usePodName bool) (bool, error) {
```

Comment:
this is a bit confusing because we always use the Pod name to resolve the Pod and get its IP address

I think something like `usePodCache` (the invert of your current boolean parameter) would be more appropriate. Even if this is test code, I think an extra function comment about this would be useful too (to tie it to CustomProbes and why the cache is not up-to-date for these temporary Pods)

Reply:
Actually for the 9 Pods in the framework that's created at the start of test, the Pod IP is not retrieved through Pod name. I also assumed so, that's why it took me quite a while to realize why my testcase was failing.

If you look closely to the `getPodsUncached` function, it retrieves Pod by listing Pods in that Namespace which has label `pod=name`. This is quite a hack... for all 9 Pods, each Pod has a label `pod=a`, `pod=b` at creation. The actual Pod name will not be x/a because the Pods are created using a deployment...

Now for the temp Pods, we used to get around this problem by providing the temp Pod with label `pod=temp-pod-name`. However, it won't work for my testcase since to test Pod add for Service, the temp Pod needs the same label to be selected by the tested Service. 

Let me know if I clarified myself.

Reply:
> However, it won't work for my testcase since to test Pod add for Service, the temp Pod needs the same label to be selected by the tested Service.

I think I understand this part. However, it seems it would have been possible to use 2 different labels, one for the name (`pod`) and one used for service selection (e.g. `app`)?

Reply:
Okay let me explore that possibility.

Reply:
Done.